### PR TITLE
Local storage as storage service

### PR DESCRIPTION
**Pull request by hero101**
- Added `LocalStorageAdapter` to use local disk as storage
- `LOCAL_STORAGE_PATH` variable maps the storage directory. Missing dirs in path will be recursively created if permissions are sufficient.
- Uses `sha3-256` hash to name and compare files.

### CODE REVIEW

File: src/common/utils/untildify.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+const homeDir = os.homedir();
+
+/***
+ * @description: Replaces '~' at the start of the path and following '/' or '\' with home
+ * @param {string} path
+ * @return {string} path
+ */
+export const untildify = (path: string): string => {
+  return homeDir ? path.replace(/^~(?=$|\/|\\)/, homeDir) : path;
```
Comment (valentinyanakiev) [inline src/common/utils/untildify.ts:11]: Will this work on Windows?

**Comment by valentinyanakiev** [comment]
[Nest] 371473  - 01/12/2024, 9:03:03 AM   ERROR [ExceptionHandler] EACCES: permission denied, mkdir '/home/valentin/alkemio/data/storage'
Error: EACCES: permission denied, mkdir '/home/valentin/alkemio/data/storage'
    at mkdirSync (node:fs:1349:3)
    at LocalStorageAdapter.ensureStoragePathExists (/home/valentin/work/repos/alkemio/server/src/services/adapters/storage/local-storage/local.storage.adapter.ts:106:16)
    at new LocalStorageAdapter (/home/valentin/work/repos/alkemio/server/src/services/adapters/storage/local-storage/local.storage.adapter.ts:28:10)
    at Injector.instantiateClass (/home/valentin/work/repos/alkemio/server/node_modules/@nestjs/core/injector/injector.js:330:19)
    at callback (/home/valentin/work/repos/alkemio/server/node_modules/@nestjs/core/injector/injector.js:48:41)
    at Injector.resolveConstructorParams (/home/valentin/work/repos/alkemio/server/node_modules/@nestjs/core/injector/injector.js:122:24)
    at Injector.loadInstance (/home/valentin/work/repos/alkemio/server/node_modules/@nestjs/core/injector/injector.js:52:9)
    at Injector.loadProvider (/home/valentin/work/repos/alkemio/server/node_modules/@nestjs/core/injector/injector.js:74:9)
    at async Promise.all (index 3)
    at InstanceLoader.createInstancesOfProviders (/home/valentin/work/repos/alkemio/server/node_modules/@nestjs/core/injector/instance-loader.js:44:9)
    at /home/valentin/work/repos/alkemio/server/node_modules/@nestjs/core/injector/instance-loader.js:29:13
    at async Promise.all (index 34)
    at InstanceLoader.createInstances (/home/valentin/work/repos/alkemio/server/node_modules/@nestjs/core/injector/instance-loader.js:28:9)
    at InstanceLoader.createInstancesOfDependencies (/home/valentin/work/repos/alkemio/server/node_modules/@nestjs/core/injector/instance-loader.js:18:9)
    at /home/valentin/work/repos/alkemio/server/node_modules/@nestjs/core/nest-factory.js:96:17
    at Function.asyncRun (/home/valentin/work/repos/alkemio/server/node_modules/@nestjs/core/errors/exceptions-zone.js:22:13)

**Comment by valentinyanakiev** [comment]
There are still a lot of IPFS_UPLOAD_FAILED etc exceptions in the code - this exception should no longer be relevant after the migration.

**Comment by valentinyanakiev** [comment]
Uploading a new card banner: 

"extensions":{2 items
"code":"IPFS_UPLOAD_FAILED"
"exception":{4 items
"message":"Ipfs upload of newFile.jpg on visual failed! Error: Uploading newFile.jpg failed!"
"context":"ipfs"
"details":{...}1 item
"stacktrace":[4 items
0:"GraphQLError: Ipfs upload of newFile.jpg on visual failed! Error: Uploading newFile.jpg failed!"
1:" at new IpfsUploadFailedException (/home/valentin/work/repos/alkemio/server/src/common/exceptions/ipfs/ipfs.upload.exception.ts:8:5)"
2:" at StorageBucketService.uploadImageOnVisual (/home/valentin/work/repos/alkemio/server/src/domain/storage/storage-bucket/storage.bucket.service.ts:339:13)"
3:" at StorageBucketResolverMutations.uploadImageOnVisual (/home/valentin/work/repos/alkemio/server/src/domain/storage/storage-bucket/storage.bucket.resolver.mutations.ts:86:28)"
]

File: src/common/utils/untildify.ts (hunk 1 above)
Comment (hero101) [reply src/common/utils/untildify.ts:11]: It will work on every OS that has `os.homedir()` set correctly.

**Comment by hero101** [comment]
@valentinyanakiev was not able to reproduce the exception, all my uploads work. The improved exception handling will shed some light on your issue.

**Review by Comoque1 — APPROVED** [review]

**Pull request merged**